subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/IntegrationTestBuildContext.java

```diff
@@ -72,6 +72,10 @@ public GradleVersion getVersion() {
         return GradleVersion.current();
```

Comment:
See comment above. The `integTest.distZipVersion` is not the version of the `testBinZip` or `intTestImage` which is used by basically all integration tests. If all integration tests would use this property here we would be fine and could set it to anything we want. But we have many integration tests which use `GradleVersion.current()` and assume the `build-receipt.properties` on their classpath and the distribution under test have the same version. This is why we need to ensure that this is the case (without the need to rework many of our tests). And this is why returning `GradleVersion.current()` is actually correct in this place. I should probably add a clarification to the getter, though.